src/appmixer/google/drive/NewFile/NewFile.js

```diff
@@ -29,6 +29,46 @@ const getNewFiles = async (lock, drive, folder, pageToken, newFiles = []) => {
     return { newFiles, newStartPageToken };
 };
 
+const detectNewFiles = async function(context) {
+
+    const { folder = {} } = context.properties;
+
+    let lock = null;
+    try {
+        lock = await context.lock(context.componentId, { maxRetryCount: 0 });
+    } catch (err) {
+        await context.stateSet('hasSkippedMessage', true);
+        return context.response();
+    }
+
+    try {
+        const { startPageToken, processedFiles = [] } = await context.loadState();
+        const auth = commons.getOauth2Client(context.auth);
+        const drive = google.drive({ version: 'v3', auth });
+
+        const { newFiles, newStartPageToken } = await getNewFiles(lock, drive, folder.id, startPageToken);
+
+        await context.stateSet('hasSkippedMessage', false);
+
+        const processedFilesSet = commons.processedItemsBuffer(processedFiles);
+
+        for (let file of newFiles) {
+            if (!processedFilesSet.has(file.id)) {
+                processedFilesSet.add(newStartPageToken, file.id);
+                await context.sendJson(file, 'file');
```

Comment:
This is something that wasn't even in the previous version, so it does not need to be done now. But, the code as it is, can create duplicated messages per `file`. Imagine that `newFiles` contains 200 files, and you send them one by one to the output port. 100 have been sent and now the system crashes. The `message` that triggered this, will be processed again and the component will be called again and it will continue from the last `startPageToken`. So the 100 files will get duplicated and sent 2x to the output port. 
Would be more bulletproof to update the state after each `sendJson`. In that case (a crash) only 1 file would get duplicated.

Reply:
it has to be here, otherwise it will duplicate the files on output regularly - 

check this [screenshot](https://private-user-images.githubusercontent.com/927250/306077942-867e280e-2584-4e99-bf0a-0ccb547e751f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkzMDkxODEsIm5iZiI6MTcwOTMwODg4MSwicGF0aCI6Ii85MjcyNTAvMzA2MDc3OTQyLTg2N2UyODBlLTI1ODQtNGU5OS1iZjBhLTBjY2I1NDdlNzUxZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwMVQxNjAxMjFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03ZDViYjJhMjc0OTU0MWVmYzJlZTE5OTVhYzhmOGIxYmIwNGNjYTNjMzQ1MzQ1YTAzYzNjMDc1ZTU0NjI4MGRjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.DDidjR-ZgRY8HYYpQ0tVoZk-QeJti1G_O_sdxbQETGQ)  from the https://github.com/clientIO/appmixer-components/pull/1696

It's quite usual the file is included in 2 different pageTokens

Reply:
good point with the updating the state, I'll update this